### PR TITLE
Fly-by cleanups

### PR DESCRIPTION
**Pull request by gnzlbg**
I wanted to add a couple of features to the crate so that we can use it in libc, and ended up doing some unrelated changes to the implementation of those features.

This PR refactors a bit the testing code, fixes the warnings in the tests with the latest nightly, adds OSX testing to travis and fixes the OSX tests (it should be easy to add a windows build bot to the crate in the future), it updates all the dependencies to the latest version (i've opened #72 to enable dependabot for the crate so that this happens automatically), fixes all "issues" discovered by clippy and adds clippy to CI, and also runs rustfmt and adds it to CI.

### CODE REVIEW

**Comment by ibabushkin** [comment]
I'll have a look tonight or tomorrow (and check the PRs from the bot as well). Thank you for you work!

**Comment by gnzlbg** [comment]
No stress, this PR does not block anything. 

There is a commit in this PR that already updates all dependencies and the affected code as necessary. The bot automatically closes its own PRs if dependencies are updated on master in some other way.

**Comment by gnzlbg** [comment]
@ibabushkin i've cherry picked here one commit that also tests rust-semverver on windows. I had to fix some `Path` handling throughout the library and the tests to make it happen, the commit shouldn't be hard to follow but let me know if you have questions.

I've rebased all my other future PRs on top of this (i want to make sure that these work at least in the tier 1 platforms), so submitting those PRs will be blocked in landing some sort of version of this.

**Comment by gnzlbg** [comment]
@ibabushkin any chance to review any of this any time soon?

**Comment by ibabushkin** [comment]
Alright, merged :) Thank you very much!
